### ISSUE TITLE
Project 2 - Univariate modeling for each feature

### ISSUE THREAD

**Issue by z3339253** [z3339253/CollabPDChallenge2017#11]
**Useful links**

http://scikit-learn.org/stable/modules/feature_selection.html
http://blog.datadive.net/selecting-good-features-part-i-univariate-selection/

**Comment by z3339253** [comment]
@Yuqianbj Hi Yuqian,

I notice that our names are under clustering for m-power dataset.  I'm confused, we don’t have access to mPower data and also not familiar with it. I think you put it the names by mistake under m-power. 

Please contact the group leader Maria and let her know you are going to work on the LDopa Univariate and ask her to omit the names. 

BTW, the deadline for this 25th Feb. We need to do some univariate modeling by then.

**Comment by z3339253** [comment]
Hi JJ,

I talked with Maria some days ago. And they are working on the univariate analysis for L-Dopa and are making a code that collects AUC (with confidence interval) and regression coefficient (with its p-value) for each feature.And trying to do this similarly to the final scoring in the competition, i.e. not all samples are used, but filtering based on age and some other factors is done. And she asked me if I have any plans, and they would like to send me their results if we like to apply the same/similar-ish function? 

And for me I'm trying to do these with our results to get familiar with these techniques. And then maybe we need to work on the features extracted by them?

Thanks,
Yuqian

**Comment by z3339253** [comment]
Hi Yuqian,

You need to perform the same on the L-Dopa dataset. 

Please  change the parameters in the R script where mpower data is supplied.

Let me know if you have any specific problems. Please also feel free to contact AP if you need any help. 

I think we have presentations today, would you be attending ? It’s 2AM for us so I don’t think I can do it.

Hi AP,

Thanks for your help. Please keep me in the loop.


-JJ

**Comment by a1png** [comment]
The code for m-power may not be useful in our work:
1. the code used linear model with logistic regression to predict, but this can only be used with binary classification, for our work, there are 4 classes(score 0-4).  Maybe we can use softmax for this? I can do it with python by not sure if it can be done with R.
2. the roc they used is also for binary classification, we also need multi class roc. this can only be done with R.

**Comment by z3339253** [comment]
yep sounds good to me, please go ahead.

**Comment by a1png** [comment]
Used python to rewrite the code
logistic regression and AUC used according to sklearn docs
[logistic regression](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression)
[roc](http://scikit-learn.org/stable/auto_examples/model_selection/plot_roc.html)

**Comment by a1png** [comment]
[univariate_tremor.csv.zip](https://github.com/z3339253/CollabPDChallenge2017/files/1994265/univariate_tremor.csv.zip)

**Comment by a1png** [comment]
Also I tried weka it can give more detailed information but I don't know if there is a way to loop through all 85 features one by one and store the results?

**Comment by z3339253** [comment]
You can post on weka forums or stackoverflow. http://weka.8497.n7.nabble.com

**Comment by z3339253** [comment]
- Uni-variate analysis using LR in python (sklearn) code was passed to Yuqian.
